state/storage.go

```diff
@@ -1690,6 +1690,20 @@ type StorageConstraints struct {
 	Count uint64 `bson:"count"`
 }
 
+// StorageDirectivesUpdate contains the user-specified constraints for updating
+// storage directives for an application unit.
+type StorageDirectivesUpdate struct {
+	// Pool is the name of the storage pool from which to provision the
+	// storage instances.
+	Pool string
```

Comment:
Use a pointer here too. Even though empty string can mean the same thing, we don't want to conflate that and we want consistency with the other attrs